src/Datadog.Trace/TracingProcessManager.cs

```diff
@@ -146,25 +150,23 @@ private static void InitializePortManagerClaimFiles(string traceAgentDirectory)
                 Directory.CreateDirectory(portManagerDirectory);
             }
 
-            var currentProcess = Process.GetCurrentProcess();
-            _processName = currentProcess.ProcessName.ToLowerInvariant();
-            _processId = currentProcess.Id;
-            var fileClaim = Path.Combine(portManagerDirectory, _processId.ToString());
+            var fileClaim =
+                Path.Combine(
+                    portManagerDirectory,
+                    $"{DomainMetadata.ProcessId}_{DomainMetadata.AppDomainId}");
 
             var portManagerFiles = Directory.GetFiles(portManagerDirectory);
-            var deleted = 0;
             if (portManagerFiles.Length > 0)
             {
                 var activePids = Process.GetProcesses().Select(p => p.Id.ToString()).ToList();
                 foreach (var portManagerFileName in portManagerFiles)
                 {
                     try
                     {
-                        var claimPid = Path.GetFileName(portManagerFileName);
-                        if (!activePids.Contains(claimPid))
+                        var claimPid = GetProcessIdFromFileName(portManagerFileName);
+                        if (claimPid == null || !activePids.Contains(claimPid))
```

Comment:
Depending on how many times we expect to call `activePids.Contains()`, it may be worth using a data structure that is optimized for search instead of `List<string>`.